### ISSUE TITLE
Check on target being between 0-1 for binary cross entropy loss

### ISSUE THREAD

**Issue by dinkarjuyal** [pytorch/pytorch#87373]
### 📚 The doc issue

In the docstring  of `https://pytorch.org/docs/stable/_modules/torch/nn/functional.html#binary_cross_entropy` it is mentioned that `target: Tensor of the same shape as input with values between 0 and 1.` However there is no such check added anywhere. 

### Suggest a potential alternative/fix

Add a check similar to the one for the input [here](https://github.com/pytorch/pytorch/blob/d56017a14f34b5130fa70c0cba010e3d2506deb0/aten/src/ATen/native/Loss.cpp#L236-L252)

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

**Comment by albanD** [comment]
We would be happy to accept a PR adding such a check to the cpu kernel you linked!
